tests/testthat/test-parser.R

```diff
@@ -70,3 +70,37 @@ test_that("Calling parse_gitlog with incorrect git repo path returns correct err
   })
 })
 
+
+# A repo with 3 commits, the first adds hello.R, the second renames it to hi.R, the last adds bye.R
+test_that("Git repo with 3 commits, first file is hello.R, then renamed to hi.R, then adds another file bye.R", {
+  # Create a temporary directory for the Git repository
+  tools_path <- file.path(tools_path)
+  tool <- yaml::read_yaml(tools_path)
+  perceval_path <- tool[["perceval"]]
+  git_repo_path <- suppressWarnings(example_repo_three_commits())
+  result <- parse_gitlog(perceval_path, git_repo_path)
+  # expect only 1 because prefix test files and suffix example files are ignored.
+  expect_equal(nrow(result), 3)
+  suppressWarnings(git_delete_sample_log(git_repo_path))
+
+})
+
+
+
+
+# A repo with 3 commits, where 1 file has as prefix _test.R, 1 file has the suffix example_*.R,
+# and 1 file hello.R.
+test_that("Git repo with 3 commits, first file has prefix:test, second file has suffix:example and last file normal", {
+  # Create a temporary directory for the Git repository
+  tools_path <- file.path(tools_path)
+  tool <- yaml::read_yaml(tools_path)
+  perceval_path <- tool[["perceval"]]
+  git_repo_path <- suppressWarnings(example_repo_suffix_prefix())
+  result <- parse_gitlog(perceval_path, git_repo_path)
+# expect only 1 because prefix test files and suffix example files are ignored.
```

Comment:
They are not ignored automatically. You still need to add filter functions.

Reply:
Should the filter functions be added in the git.R file?

Reply:
On my more recent commit, I already addressed that. 

Just to be clear: The code revisions here were added to help you understand why I started making commits to fix your checks not passing. My subsequent commits addressed a number of them. 

So, check that the more recent version of my commits fixes the issue I raised, then compare to see how I addressed them. You will want to understand this well for milestone 2. 

Note I did not mention exhaustively everything you have to change on this code review. Some you will need to inspect the code yourself, and test the line execution to make sure it all work as intended. 

For example, I may not have used the new function to delete folder in all your examples, there are still functions poorly documented or without title, etc.